### PR TITLE
Add Support for Elementwise Logical_Not Ops

### PR DESCRIPTION
**Pull request by Atrisan**
## Description
Adds a new library node for logical not operations on integers and floats. 
Adds support for the aten.logical_not operation to the pytorch frontend.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal change (no functional change)
- [ ] Documentation update

## Quality Checklist
*Please ensure the following are completed before requesting review:*

* [x] Runtime-compatible (no externally visible existing method has its parameters changed, was deleted or has its exception behavior changed)
* [x] Compile-time compatible (new arguments & properties added have default values, but users need to recompile against the new version to work)

IR:
* [x] Introduces new types into the SDFG IR
* [ ]  Changed / added properties on SDFG IR Elements (LibraryNodes, ControlFlowNodes DataFlowNodes etc.)
  *  [ ] New / modified types have serialization / deserialization support
  *  [ ] Deserialization is backward compatible (new serialization can deserialize old format and represent it in the new format if applicable)
  *  [ ] Serialized format is backward compatible (previous deserializer can read new serialized format without incorrectness (just lacking additional guarantees that are optional or did not exist before)
  *  [ ] Serializer has an option to emit the previous format

### Unit Tests
- [x] New unit tests have been added for the changes.
- [x] Existing unit tests pass locally.
- [x] Edge cases and error paths are covered.

### Integration Tests
- [x] New integration tests have been added (or existing ones updated).
- [x] Integration tests pass locally.
- [ ] Interactions with other components/services are verified.

### Documentation
- [ ] Public APIs, options, and behavior changes are documented.
- [ ] README / docs site / inline comments updated where applicable.
- [ ] Changelog / release notes updated (if applicable).

### Test Coverage
- [ ] Test coverage has not decreased as a result of this change.
- [ ] Coverage report reviewed and acceptable.
- [ ] Critical / new code paths are covered.

## How Has This Been Tested?
Describe the tests you ran to verify your changes and provide instructions to reproduce.
Include relevant details for your test configuration (OS, Python/LLVM/Docc version, hardware/target).

## Additional Context
Add any other context about the PR here (design decisions, trade-offs, follow-ups).

### CODE REVIEW

File: sdfg/include/sdfg/data_flow/library_nodes/math/tensor/elementwise_ops/logical_not_node.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include "sdfg/codegen/dispatchers/block_dispatcher.h"
+#include "sdfg/serializer/json_serializer.h"
+
+namespace sdfg {
+namespace math {
+namespace tensor {
+
+inline data_flow::LibraryNodeCode LibraryNodeType_LogicalNot("ml::LogicalNot");
+
+class LogicalNotNode : public ElementWiseDataflowTensorNode {
```
Comment (ramonwirsch) [inline sdfg/include/sdfg/data_flow/library_nodes/math/tensor/elementwise_ops/logical_not_node.h:14]: Documentation

**Review by ramonwirsch — APPROVED** [review]

**Review by Moehre2 — APPROVED** [review]
